CDP4Reporting/DataSource/ReportingDataSourceCategory.cs

```diff
@@ -0,0 +1,104 @@
+//    version 3 of the License, or any later version.
+//
+//    The CDP4-IME Community Edition is distributed in the hope that it will be useful,
+//    but WITHOUT ANY WARRANTY; without even the implied warranty of
+//    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//    GNU Affero General Public License for more details.
+//
+//    You should have received a copy of the GNU Affero General Public License
+//    along with this program. If not, see <http://www.gnu.org/licenses/>.
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace CDP4Reporting.DataSource
+{
+    using System.Linq;
+
+    using CDP4Common.CommonData;
+    using CDP4Common.EngineeringModelData;
+    using CDP4Common.SiteDirectoryData;
+
+    /// <summary>
+    /// Abstract base class from which all category columns for a <see cref="ReportingDataSourceRow"/> need to derive.
+    /// </summary>
+    public abstract class ReportingDataSourceCategory<T> : ReportingDataSourceColumn<T> where T : ReportingDataSourceRow, new()
+    {
+        /// <summary>
+        /// Gets or sets the associated <see cref="Category"/> short name.
+        /// </summary>
+        public string ShortName { get; private set; }
+
+        /// <summary>
+        /// Flag indicating whether the associated <see cref="Category"/> is present.
+        /// </summary>
+        public bool Value { get; private set; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ReportingDataSourceCategory{T}"/> class.
+        /// </summary>
+        protected ReportingDataSourceCategory()
+        {
+            this.ShortName = GetParameterAttribute(this.GetType())?.ShortName;
+        }
+
+        /// <summary>
+        /// Initializes a reported category column based on the corresponding <see cref="ElementBase"/>
+        /// within the associated <see cref="ReportingDataSourceNode{T}"/>.
+        /// </summary>
+        /// <param name="node">
+        /// The associated <see cref="ReportingDataSourceNode{T}"/>.
+        /// </param>
+        internal override void Initialize(ReportingDataSourceNode<T> node)
+        {
+            this.Node = node;
+            this.Value = this.IsMemberOfCategory();
+        }
+
+        /// <summary>
+        /// Checks if the <see cref="ElementDefinition.Category"/>, or <see cref="ElementUsage.Category"/> contains a <see cref="Category"/>
+        /// that has <see cref="ReportingDataSourceCategory{T}.ShortName"/> as its <see cref="Category.ShortName"/>, or somewhere in the
+        /// <see cref="Category.SuperCategory"/> hierarchy.
+        /// </summary>
+        /// <returns>
+        /// True if <see cref="Category.ShortName"/> was found, otherwise false.
+        /// </returns>
+        private bool IsMemberOfCategory()
+        {
+            if (this.Node.NestedElement == null)
```

Comment:
Can we have nodes (rows) that are not associated to an element?

Reply:
At the moment we cannot have node rows that are not associated to an element.
So scoping functionality and names of classes is hard. We want to stay backwards compatible with reporting in the future, so every class that can be used in the DataSource C# editor should stay backwards compatible.
It is all related and I think we should have a discussion on this.

Reply:
> At the moment we cannot have node rows that are not associated to an element.
Then the `if` statement can be removed.

The naming of classes is indeed a bigger discussion.